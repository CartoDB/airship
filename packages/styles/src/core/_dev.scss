@import 'tools/_mixins';
@import 'variables/_variables';
