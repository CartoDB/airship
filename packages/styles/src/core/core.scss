@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
