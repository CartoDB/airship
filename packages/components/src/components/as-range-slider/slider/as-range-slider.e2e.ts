import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-range-slider', () => {
  let page: E2EPage;

  beforeEach(async () => {
    page = await newE2EPage();

    await page.setContent(`
      <as-range-slider></as-range-slider>
    `);
  });

  describe('Rendering', () => {
    it('should render without parameters, using defaults', async () => {
      await page.waitForChanges();

      const element = await page.find('as-range-slider');
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should render 1 thumb + 1 bar when using a value', async () => {
      const element = await page.find('as-range-slider');
      element.setProperty('value', 5);
      element.setProperty('minValue', 0);
      element.setProperty('maxValue', 10);
      element.setProperty('step', 1);
      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should render 2 thumbs + 1 bar when using a range', async () => {
      const element = await page.find('as-range-slider');
      element.setProperty('range', [2, 4]);
      element.setProperty('minValue', 2);
      element.setProperty('maxValue', 20);
      element.setProperty('step', 2);
      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('can be rendered disabled', async () => {
      const element = await page.find('as-range-slider');
      element.setProperty('disabled', true);
      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
