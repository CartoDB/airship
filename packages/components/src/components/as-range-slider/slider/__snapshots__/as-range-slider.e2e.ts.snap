// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-range-slider Rendering can be rendered disabled 1`] = `"<as-range-slider class=\\"hydrated\\"><div class=\\"as-range-slider as-range-slider--disabled\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"-1\\" aria-valuetext=\\"0\\" aria-valuenow=\\"0\\" aria-valuemin=\\"0\\" aria-valuemax=\\"10\\" class=\\"as-range-slider__thumb as-range-slider__thumb--disabled\\" data-value=\\"0\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value as-range-slider__value--disabled\\">0</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar as-range-slider__range-bar--disabled\\"></div></as-range-slider-bar></div></div></as-range-slider>"`;

exports[`as-range-slider Rendering should render 1 thumb + 1 bar when using a value 1`] = `"<as-range-slider class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"5\\" aria-valuenow=\\"5\\" aria-valuemin=\\"0\\" aria-valuemax=\\"10\\" class=\\"as-range-slider__thumb\\" data-value=\\"5\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value\\">5</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\"></div></as-range-slider-bar></div></div></as-range-slider>"`;

exports[`as-range-slider Rendering should render 2 thumbs + 1 bar when using a range 1`] = `"<as-range-slider class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"2\\" aria-valuenow=\\"2\\" aria-valuemin=\\"0\\" aria-valuemax=\\"3\\" class=\\"as-range-slider__thumb\\" data-value=\\"2\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value\\">2</span></div></as-range-slider-thumb><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"4\\" aria-valuenow=\\"4\\" aria-valuemin=\\"3\\" aria-valuemax=\\"10\\" class=\\"as-range-slider__thumb\\" data-value=\\"4\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value\\">4</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\"></div></as-range-slider-bar></div></div></as-range-slider>"`;

exports[`as-range-slider Rendering should render without parameters, using defaults 1`] = `"<as-range-slider class=\\"hydrated\\"><div class=\\"as-range-slider\\"><div class=\\"as-range-slider__rail\\"><as-range-slider-thumb class=\\"hydrated\\"><div role=\\"slider\\" tabindex=\\"0\\" aria-valuetext=\\"0\\" aria-valuenow=\\"0\\" aria-valuemin=\\"0\\" aria-valuemax=\\"10\\" class=\\"as-range-slider__thumb\\" data-value=\\"0\\"><div class=\\"as-range-slider__thumb-handle\\"></div><span class=\\"as-caption as-font-medium as-range-slider__value\\">0</span></div></as-range-slider-thumb><as-range-slider-bar class=\\"hydrated\\"><div class=\\"as-range-slider__range-bar\\"></div></as-range-slider-bar></div></div></as-range-slider>"`;
